docs/debugging.md

```diff
@@ -196,3 +196,100 @@ will ensure that the source files are kept as well, such files will be loaded
+CI instead of compiling Cython, which can take up to a few minutes on some CI
+runners.
+
+### Cython compilation errors arising from `static_assert`
+
+This error arises from a limitation in Cython that exposes a CPython
+implementation detail. Under clang, the issue manifests in error messages like
+the following example, generated by building scikit-learn with a C99 compilation
+environment:
+
+```
+      In file included from sklearn/__check_build/_check_build.cpython-313t-darwin.so.p/sklearn/__check_build/_check_build.pyx.c:4315:
+      In file included from /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_frame.h:13:
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_code.h:537:15: error: expected parameter declarator
+      static_assert(COLD_EXIT_INITIAL_VALUE > ADAPTIVE_COOLDOWN_VALUE,
+                    ^
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_backoff.h:125:33: note: expanded from macro 'COLD_EXIT_INITIAL_VALUE'
+      #define COLD_EXIT_INITIAL_VALUE 64
+                                      ^
+      In file included from sklearn/__check_build/_check_build.cpython-313t-darwin.so.p/sklearn/__check_build/_check_build.pyx.c:4315:
+      In file included from /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_frame.h:13:
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_code.h:537:15: error: expected ')'
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_backoff.h:125:33: note: expanded from macro 'COLD_EXIT_INITIAL_VALUE'
+      #define COLD_EXIT_INITIAL_VALUE 64
+                                      ^
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_code.h:537:14: note: to match this '('
+      static_assert(COLD_EXIT_INITIAL_VALUE > ADAPTIVE_COOLDOWN_VALUE,
+                   ^
+      /path/to/python/3.13.0b3t/include/python3.13t/internal/pycore_code.h:537:1: error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
+      static_assert(COLD_EXIT_INITIAL_VALUE > ADAPTIVE_COOLDOWN_VALUE,
+      ^
+      int
+```
+
+Freethreading support in Cython relies CPython internals. In particular, cython
+will in some generate code that in some situations includes an internal CPython
+header, `pycore_frame.h`. This header, in turn, includes another header
+`pycore_code.h` that makes use of `static_assert`, a construct defined in the
+C11 standard. This is problematic for a compiler assuming compliance with
+C99. 
+
+In practice we have found the easiest way to fix this is to update the standard
+used to build C code to either C11 or C17. In the meson build configuration for
+a simple C project, this looks like:
+
+```meson
+project(
+  'my-project',
+  'c',
+  default_options: [
+    'c_std=c17',
+  ],
+)
+```
+
+Note that C17 is for the most part a subset of C11, so requiring C11 support may
```

Comment:
"subset of C11" -> "bug-fix release for C11" or something like that? I don't think anything got removed.